OpenDreamRuntime/Objects/Types/DreamList.cs

```diff
@@ -77,6 +84,25 @@ public override void Initialize(DreamProcArguments args) {
         }
     }
 
+    // Hold on to large lists for reuse later
+    protected override void HandleDeletion(bool possiblyThreaded) {
+        if (_values.Capacity < 2048) {
+            base.HandleDeletion(possiblyThreaded);
+            return;
+        }
+
+        if (possiblyThreaded) {
+            EnterIntoDelQueue();
+            return;
+        }
+
+        base.HandleDeletion(possiblyThreaded);
+        if (ListPool.Count < 5000) {
```

Comment:
Any sort of caching like this should be a tunable value (i.e. a cvar)
May also be worth being more conservative, as is this is 163 **megabytes** of pooled lists! You probably want to not pool this many...